spec/syslogger_spec.rb

```diff
@@ -3,9 +3,51 @@
 describe "Syslogger" do
 
   expected_info_level = Syslog::LOG_NOTICE
+  test_formatter = Proc.new { |severity, time, progname, message| message }
+
+  describe "formatter" do
+    before(:each) do
+      @logger = Syslogger.new("my_app", Syslog::LOG_PID | Syslog::LOG_CONS, nil)
+    end
+
+    it 'is applied when set' do
+      @logger.formatter = Proc.new { 'success' }
+      expect(Syslog).to receive(:open).and_yield(syslog=double('syslog', :mask= => true))
+
+      expect(syslog).to receive(:log).with(Syslog::LOG_WARNING, 'success')
+
+      @logger.warn('test message')
+    end
+
+    it 'can be read back' do
+      @logger.formatter = test_formatter
+
+      expect(@logger.formatter).to be(test_formatter)
+    end
+
+    it 'is applied to the message, all params passed' do
+      expect(Syslog).to receive(:open).and_yield(syslog=double('syslog', :mask= => true))
+      expect(syslog).to receive(:log)
+      args = nil
+      @logger.formatter  = Proc.new { |severity, time, progname, message|
+        args = { severity: severity, time: time, progname: progname, message: message }
+      }
+      @logger.ident = 'my-prog'
+
+      before = Time.now
+      @logger.warn('my-message')
+      after = Time.now
+
+      expect(args[:severity]).to eq('WARN')
+      expect(args[:time]).to (be >= before).and(be <= after)
+      expect(args[:progname]).to eq('my-prog')
+      expect(args[:message]).to eq('my-message')
```

Comment:
BTW, if you want to be explicit about the `between` being inclusive, you can chain that off of it:

``` ruby
a_value_between(before, after).inclusive
```

It's inclusive by default, though so it's not necessary. `exclusive` is also supported.

For future reference, the best place to find built-in RSpec matchers is the list of instance methods here:

http://rspec.info/documentation/3.2/rspec-expectations/RSpec/Matchers.html

`a_value_between` is an alias of `be_between` that reads like a noun for situations when you are passing it to another matcher:

http://rspec.info/documentation/3.2/rspec-expectations/RSpec/Matchers.html#be_between-instance_method